pennylane/hf/tapering.py

```diff
@@ -464,3 +467,106 @@ def transform_hamiltonian(h, generators, paulix_ops, paulix_sector):
     c = qml.math.stack(c)
 
     return _simplify(qml.Hamiltonian(c, o))
+
+
+def _sector_energy(sector):
+    """Computes energy of the given tapered Hamiltonian.
+
+    Args:
+        sector (pennylane.Hamiltonian): tapered Hamiltonian whose energy has to be calculated
+
+    Returns:
+        energy (float): energy of the tapered hamiltonian
+    """
+
+    if len(sector.wires) < 2:
+        energy = min(sp.linalg.eigvals(qml.utils.sparse_hamiltonian(sector).toarray()))
+    else:
+        energy = sp.sparse.linalg.eigs(qml.utils.sparse_hamiltonian(sector), k=2)[0].min()
+
+    return energy
+
+
+def find_optimal_sector(qubit_op, generators, paulix_ops, brute_force=True, num_electrons=0):
+    r"""Get the optimal sector which contains the ground state.
+
+    To obtain the optimal sector one can brute force through all the permutation or by utilize the following
+    relation between the Pauli-Z qubit operator and the occupation number under Jordan-Wigner transform.
+
+    .. math::
+
+        \sigma_z^{i} = a_{i}^{\dagger}a_{i} - 1
```

Comment:
Shouldn't this be \sigma_z^{i} = 1 - 2 a_{i}^{\dagger}a_{i} or maybe I am missing something?

Reply:
Number operator is `[[0, 0], [0, 1]]` so indeed the formula Soran is quoting should be the correct one. Also, I think it's better to express the number operator in terms of the Pauli operators. I suggest changing to

`n_i=a_{i}^{\dagger}a_{i}=I-2\sigma_z^i`

But please check that we are using labels on the Pauli operators consistently. I recall in other PRs we were using `\sigma_i^z`, though I may be wrong

Reply:
Updated. Even I'd gotten the same result when I was calculating it by myself. But I wonder why [Setia's paper](https://arxiv.org/pdf/1910.14644.pdf)  that we are following used this definition (Eq 17).